### PR TITLE
modified server code to send formatted json to front end

### PR DESCRIPTION
**Pull request by alexander871015**
The json should have the format of {dataset : [{date : status}, ...]} where dataset is a string of the name and date is string representation of the date, and status is a boolean.

### CODE REVIEW

File: hologram_dashboard/server/request_handler/request_handler.cc [hunk 1]
```diff
@@ -38,34 +38,30 @@ void RequestHandler::GetDashboard(const Pistache::Rest::Request& request,
...
-        stream << "\t<tr>\n";
-        stream << "\t\t<td>" << data.c_str() << "&nbsp&nbsp</td>\n";
-        for(const std::string& date : dates) {
-            stream << "\t\t<td class=";
-            if (status[i++] == '1') {
-                stream << "\"online\"";
+    auto stream = response.stream(Pistache::Http::Code::Ok);
+    if (system_in == "Chipper") {
+        int status_idx = 0;
+        for (const std::string& data : dataSources) {
```
Comment (emilyqu) [inline hologram_dashboard/server/request_handler/request_handler.cc:48]: nit: for (const std::string& data_source : data_sources)

File: hologram_dashboard/server/request_handler/request_handler.cc [hunk 2]
```diff
@@ -38,34 +38,30 @@ void RequestHandler::GetDashboard(const Pistache::Rest::Request& request,
...
+            message[data] = nlohmann::json::array();
+            for (const std::string& date : dates) {
+                message[data].push_back({date, status[status_idx++] == '1'});
             }
```
Comment (emilyqu) [inline hologram_dashboard/server/request_handler/request_handler.cc:52]: could you explain what `1` means? available?

File: hologram_dashboard/server/request_handler/request_handler.cc [hunk 3]
```diff
@@ -38,34 +38,30 @@ void RequestHandler::GetDashboard(const Pistache::Rest::Request& request,
...
-            else {
-                stream << "\"offline\"";
+        }
+    } else {
+        int status_idx = status.size() - 1;
+        for (const std::string& data : dataSources) {
+            message[data] = nlohmann::json::array();
+            for (const std::string& date : dates) {
+                message[data].push_back({date, status[status_idx--] == '1'});
             }
```
Comment (emilyqu) [inline hologram_dashboard/server/request_handler/request_handler.cc:60]: I am a bit confused about this part. why starting from status.size()-1 and go backwards?

File: hologram_dashboard/server/request_handler/request_handler.cc (hunk 2 above)
Comment (alexander871015) [reply hologram_dashboard/server/request_handler/request_handler.cc:52]: ya for this hard coded example 1 is true 0 is false, it is entirely randomly generated.

File: hologram_dashboard/server/request_handler/request_handler.cc (hunk 3 above)
Comment (alexander871015) [reply hologram_dashboard/server/request_handler/request_handler.cc:60]: just so that chipper and chipper gdpr will have different random values.

Comment (emilyqu) [reply hologram_dashboard/server/request_handler/request_handler.cc:65]: Data is hard-coded anyway, making them different value seems a bit unnecessary.
Plus, could you hard-code the data availability proto instead? Your data should keep in sync with what database would return.

**Pull request merged**